tests/RobotFramework/tests/cumulocity/monitoring/monitor_device_collectd.robot

```diff
@@ -54,7 +54,7 @@ Check grouping of measurements
     ${start_time}=    Get Unix Timestamp
     Execute Command    tedge mqtt pub collectd/localhost/temperature/temp1 "`date +%s.%N`:50" && tedge mqtt pub collectd/localhost/temperature/temp2 "`date +%s.%N`:40" && tedge mqtt pub collectd/localhost/pressure/pres1 "`date +%s.%N`:10" && tedge mqtt pub collectd/localhost/pressure/pres2 "`date +%s.%N`:20"
     ${c8y_messages}    Should Have MQTT Messages    c8y/measurement/measurements/create    maximum=4    date_from=${start_time}
-    Should Contain Any   @{c8y_messages}    "temp1":{"value":50.0}    "temp2":{"value":40.0}    "pres1":{"value":10.0}    "pres2":{"value":20.0}
+    Should Contain Any   ${c8y_messages[0]}    "temp1":{"value":50.0}    "temp2":{"value":40.0}    "pres1":{"value":10.0}    "pres2":{"value":20.0}
```

Comment:
This test is really weak. I fully appreciate that this is mainly due to the collectd mapper being non deterministic, but this raises a concern: how can the collectd mapper output be more predictable? Currently there are internal parameters that are not exposed to the user (as the size of the time-window used to group messages as well as the maximum accepted delay for a message). Should we make these configurable?

Reply:
Yes making the batching size more configurable would be very useful for testing (I think making things more testable is also a valid use-case, not just for users). Though this style of tests was a comprise due to the lack of control of the message grouping.